app/src/androidTest/java/com/android/bookswap/ui/profile/OthersUserProfileTest.kt

```diff
@@ -183,4 +183,86 @@ class OthersUserProfileTest : TestCase() {
         .onAllNodesWithTag(C.Tag.BookDisplayComp.author)
         .assertAll(hasText("Author 1").or(hasText("Author 2")))
   }
+
+  @Test
+  fun displaysImage_whenPhotoUrlIsNotEmpty() {
+    val imageTestUserId = UUID.randomUUID()
+    val imageTestUser =
+        DataUser(
+            userUUID = imageTestUserId,
+            greeting = "Mr.",
+            firstName = "John",
+            lastName = "Doe",
+            email = "john.doe@example.com",
+            phoneNumber = "1234567890",
+            latitude = 45.0,
+            longitude = 50.0,
+            profilePictureUrl = "https://test_profile_pic.jpg",
+            bookList = listOf(UUID.randomUUID(), UUID.randomUUID()))
+
+    // Mock the user data fetching
+    every { mockOthersUserViewModel.getUserByUUID(imageTestUserId, any()) } answers
+        {
+          val callback = secondArg<(DataUser?) -> Unit>()
+          callback(imageTestUser)
+        }
+
+    composeTestRule.setContent {
+      OthersUserProfileScreen(
+          userId = imageTestUserId,
+          otherUserVM = mockOthersUserViewModel,
+          booksRepository = mockBooksRepository,
+          userBookViewModel = mockUserBookViewModel,
+          navigationActions = mockNavigationActions)
+    }
+    // Verify the container is displayed
+    composeTestRule.onNodeWithTag(C.Tag.OtherUserProfile.image_container).assertIsDisplayed()
+
+    // Verify the picture is displayed inside the container
+    composeTestRule.onNodeWithTag(C.Tag.OtherUserProfile.profile_image_picture).assertIsDisplayed()
+
+    // Ensure the icon is not displayed
+    composeTestRule.onNodeWithTag(C.Tag.OtherUserProfile.profile_image_icon).assertDoesNotExist()
+  }
+
+  @Test
+  fun displaysIcon_whenPhotoUrlIsEmpty() {
+    val imageTestUserId = UUID.randomUUID()
+    val imageTestUser =
+        DataUser(
+            userUUID = imageTestUserId,
+            greeting = "Mr.",
+            firstName = "John",
+            lastName = "Doe",
+            email = "john.doe@example.com",
+            phoneNumber = "1234567890",
+            latitude = 45.0,
+            longitude = 50.0,
+            profilePictureUrl = "",
+            bookList = listOf(UUID.randomUUID(), UUID.randomUUID()))
```

Comment:
the `testUser` defined at the top can be used in this scenario instead of creating a new dataUser.